turbomoleio/core/molecule.py

```diff
@@ -511,6 +604,419 @@ def from_dict(cls, d):
+            f.write(self.to_coord_string())
+
+    def coord_lines(self):
+        """
+        Gives the lines of the $coord datagroup based on the internal Molecule.
+
+        Returns:
+            list: list of strings with the $coord datagroup.
+        """
+
+        return get_coord_lines(self.structure, self.frozen_indices)
+
+    def to_coord_string(self):
+        """
+        Creates the string of a coord file for this system.
+
+        Returns:
+            str: the string representing the system.
+        """
+        lines = ["$coord"]
+        lines.extend(self.coord_lines())
+
+        if self.periodicity is not None:
+            lines.append("$periodic {}".format(self.periodicity))
+            lattice = self.structure.lattice
+            lmat = lattice.matrix
+            if (
+                    not np.allclose(lmat[0][1:], 0.0) or
+                    not np.isclose(lmat[1][2], 0.0)
+            ):
+                raise ValueError('Lattice should be oriented such that first vector '
+                                 'is aligned with x cartesian direction and second vector '
+                                 'is in the xy cartesian plane.')
+
+            if self.periodicity == 1:
+                lines.append(f"$cell\n  {ang_to_bohr * lattice.a}")
+            elif self.periodicity == 2:
+                lines.append(f"$cell\n  "
+                             f"{ang_to_bohr * lattice.a}   "
+                             f"{ang_to_bohr * lattice.b}   "
+                             f"{lattice.gamma}")
+            elif self.periodicity == 3:
+                lines.append(f"$cell\n  "
+                             f"{ang_to_bohr * lattice.a}   "
+                             f"{ang_to_bohr * lattice.b}   "
+                             f"{ang_to_bohr * lattice.c}   "
+                             f"{lattice.alpha}   "
+                             f"{lattice.beta}   "
+                             f"{lattice.gamma}"
+                             )
+            else:
+                raise ValueError('Periodicity should be 1, 2 or 3 for use in riper calculations.')
+
+        lines.append("$end")
+
+        return "\n".join(lines)
+
+    def to_file(self, filepath, fmt=None):
+        """
+        Writes the system to a file depending on the specified format. Could be
+        a coord file or any other format supported by pymatgen Molecule.
+
+        Args:
+            filepath (str): path to the file.
+            fmt (str): the format of the data. could be "coord" for Turbomole
+                coord file or any format supported in pymatgen Molecule.
+                If None the code would try to infer it from the file name.
+        """
+        fname = os.path.basename(filepath)
+        if fmt == "coord" or (fmt is None and fnmatch(fname, "*coord*")):
+            self.to_coord_file(filepath)
+        else:
+            self.molecule.to(filename=filepath, fmt=fmt)
+
+    def _check_index(self, indices_list):
```

Comment:
internal coordinates are not well supported for periodic systems in TM, right? Should we have all these methods anyway in `PeriodicSysytem`?

Reply:
I've removed them already (had'nt seen this comment yet), see above comment.

Reply:
also the method `_check_index` seems to be only related to the internal coordinate and could be removed.